Automattic/sensei
# Migrate all e2e files to typescript

Resolves the #6286 


## #  Extra Notes:

1) I could not break into small PRs due to a combo of issues, because In order to add support to the absolute path I had to configure the eslint properly to use typescript, and when I configured the eslint I receive a lot of errors, some of the causes was try to support  (JS  + commonjs)  +  (JS + es6 modules) + (Typescript).
So to simplify I migrated all to typescript.

2) There are more opportunities to improve gradually in other PRs. Examples of improvements:
* Move fixtures to a proper place
* Move the types (Course, Lesson, etc...) to a shared place
* Separate the concern a bit better.




### Introduced changes

This PR is big but most changes are just renaming files from.js to .ts and adding types the exceptions are:
* Enable `.eslint` to check the TS files
* Enable to adopt absolute paths during importing, example [here](https://github.com/Automattic/sensei/pull/6285/files#diff-6dd18a54d34a90c6e5a0523236311870fb62a3b6de36023dedd14c2b98e8a703R11-R13) 
* Quick Refactor on the course-completion spec
* Add the script `npm run lint-e2e` to lint the e2e files

### Testing instructions
* Revert this commit in your env 887b3060171c5c68722911d7813a375f267ebac5
* run `npm install`
* run `npm run lint-e2e`
* Check if there is one error like this one
<img width="1048" alt="Screenshot 2022-12-12 at 21 35 10" src="https://user-images.githubusercontent.com/38718/207197533-baf4f637-6c9b-4dbd-89fe-6c040e5fe997.png">
* Try to run the e2e tests
* Remove the revert commit
* Run the e2e tests ` npm run test:e2e -- --grep-invert @setup`
